pyblish/plugin.py

```diff
@@ -58,6 +60,45 @@
 STRICT_DATATYPES = EARLY_ADOPTER or STRICT_DATATYPES
 
 
+class ProfilingTimer():
```

Comment:
I think we should stick with a `@contextmanager`, like it's being used elsewhere in Pyblish.

```python
import contextlib

@contextlib.contextmanager
def profiling():
  stats = {}
  stats["startTime"] = time.time()
  yield stats
  stats["endTime"] = time.time()

with profiling() as stats:
  ...

```

Alternatively..

```py
stats = {}
with profiling(stats):
  ...
```